patches/net/minecraft/world/entity/player/Player.java.patch

```diff
@@ -268,9 +268,16 @@
                      }
  
                      float f3 = f + f1;
-@@ -1179,9 +_,7 @@
+                     boolean flag2 = false;
                      double d0 = (double)(this.walkDist - this.walkDistO);
-                     if (flag4 && !flag1 && !flag && this.onGround() && d0 < (double)this.getSpeed()) {
+-                    if (flag4 && !flag1 && !flag && this.onGround() && d0 < (double)this.getSpeed()) {
++                    boolean allowSweepPreconditions = flag4 && !flag && this.onGround() && d0 < (double)this.getSpeed();
++                    // Neo: the boolean local above (allowSweepPreconditions) is the vanilla sweep attack condition, excluding the condition of not doing critical hit
++                    // Fire the sweep attack event based on current information.
++                    // The sweep attack event default result will be allowSweep && !critEvent.disableSweep()
++                    // The event cannot bypass item ability check
```

Comment:
Why is the event limited to not bypassing the item ability check? That seems like a perfectly reasonable thing for this to be able to override, especially given all the other things it's already able to bypass.

Reply:
addressed